compiler/haskell-ide-core/src/Development/IDE/Functions/Compile.hs

```diff
@@ -241,22 +241,21 @@ runGhcSession
     -> Ghc a
     -> IO a
 runGhcSession IdeOptions{..} modu env act = runGhcEnv env $ do
-    modifyDynFlags $ \x -> x{importPaths = maybe [] moduleImportPaths modu ++ importPaths x}
+    modifyDynFlags $ \x -> x
+        {importPaths = nubOrd $ maybe [] (maybeToList . moduleImportPaths) modu ++ importPaths x}
```

Comment:
```suggestion
        {importPaths = nubOrd $ maybeToList (moduleImportPaths =<< modU) ++ importPaths x}
```
codegolf at this point but it took me a while to decypher the types here so maybe this helps.

Reply:
Applied minus the typo - thanks!